frontend/admin/src/js/components/Table/Table.tsx

```diff
@@ -140,6 +142,9 @@ function Table<T extends Record<string, unknown>>({
   hiddenCols = [],
   initialSortBy = [],
 }: TableProps<T>): ReactElement {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const initialState = useInitialTableState(searchParams);
```

Comment:
We need access to the state returned by `react-table` to store the new state in the `useEffect` and since the `initialState` is used in setting up the table, it needs to be defined first so there is no way to do this in a clean way unless we move the `useTable` hook there as well.

Reply:
We could still wrap useSearchParams in `useInitialTableState()`, and expose setSearchParams as setInitialState: `const [initialState, setInitialState] = useInitialTableState();`

Reply:
This is another case of me being stumped on how to get this to work and not result in an infinite render loop. The reason to decoupling all of this was to ensure `initialState` was very stable and unchanging since any change to the initial state causes a re-render which ends up in an infinite re-render. 😕